### PR TITLE
Upgrade `@solana/wallet-adapter-react` for compatibility with Mobile Wallet Adapter and the Wallet Standard

### PR DESCRIPTION
**Pull request by steveluscher**
# Description

In this PR we upgrade `@solana/wallet-adapter-react` to a version that has the following benefits:

1. The ability to connect to _any_ mobile wallet app that supports the Solana Mobile Stack Mobile Wallet Adapter Protocol.
2. The ability to connect to _any_ desktop wallet that supports the Solana Wallet Standard.

You can read much more about the wider effort to upgrade the entire ecosystem of Solana web apps at: https://github.com/solana-labs/wallet-adapter/issues/604

### Mobile, with mobile wallet
 
https://user-images.githubusercontent.com/13243/197319831-27e439e8-2148-4944-8e25-7b4082f20405.mp4

### Desktop, with desktop wallets

https://user-images.githubusercontent.com/13243/197319836-d7f3edc0-bfd0-4075-a421-f0408a9a902d.mov

### CODE REVIEW

**Comment by steveluscher** [comment]
I did _not_ test this with an actual live Candy Machine. If you have one handy, that would be helpful!

File: src/App.tsx [hunk 1]
```diff
@@ -58,11 +58,11 @@ const App = () => {
...
-      getPhantomWallet(),
-      getSolflareWallet(),
-      getSlopeWallet(),
-      getSolletWallet({ network }),
-      getSolletExtensionWallet({ network }),
+      new PhantomWalletAdapter({ network }),
+      new SolflareWalletAdapter({ network }),
+      new SlopeWalletAdapter({ network }),
+      new SolletWalletAdapter({ network }),
+      new SolletExtensionWalletAdapter({ network }),
```
Comment (jordansexton) [inline src/App.tsx:65]: Memo is missing `network` dependency.

File: src/App.tsx [hunk 2]
```diff
@@ -58,11 +58,11 @@ const App = () => {
...
   const wallets = useMemo(
     () => [
-      getPhantomWallet(),
-      getSolflareWallet(),
-      getSlopeWallet(),
-      getSolletWallet({ network }),
-      getSolletExtensionWallet({ network }),
+      new PhantomWalletAdapter({ network }),
+      new SolflareWalletAdapter({ network }),
+      new SlopeWalletAdapter({ network }),
```
Comment (jordansexton) [inline src/App.tsx:63]: Slope's adapter doesn't take a `network` param.

File: src/App.tsx [hunk 3]
```diff
@@ -58,11 +58,11 @@ const App = () => {
 
   const wallets = useMemo(
     () => [
-      getPhantomWallet(),
-      getSolflareWallet(),
-      getSlopeWallet(),
-      getSolletWallet({ network }),
-      getSolletExtensionWallet({ network }),
+      new PhantomWalletAdapter({ network }),
```
Comment (jordansexton) [inline src/App.tsx:61]: Phantom's adapter doesn't take a `network` param.

**Review by jordansexton — COMMENTED** [review]
Some minor `network` fixes. I'm guessing the project TS checking / linting isn't very strict?

File: src/App.tsx (hunk 3 above)
Comment (steveluscher) [reply src/App.tsx:61]: Hunh. Why did TypeScript lie to me?

File: src/App.tsx (hunk 2 above)
Comment (steveluscher) [reply src/App.tsx:63]: TYPESCRIPT!!!1!one!

File: src/App.tsx (hunk 1 above)
Comment (steveluscher) [reply src/App.tsx:65]: That's because `network` is outside of the render function (ie. is constant).

**Review by steveluscher — COMMENTED** [review]
Will update!

**Comment by stegaBOB** [comment]
looks good to me!

**Pull request merged**